pkg/agentscheduler/actions/allocate/allocate.go

```diff
@@ -0,0 +1,263 @@
+/*
+Copyright 2025 The Volcano Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package allocate
+
+import (
+	"fmt"
+
+	"k8s.io/klog/v2"
+
+	agentapi "volcano.sh/volcano/pkg/agentscheduler/api"
+	"volcano.sh/volcano/pkg/agentscheduler/framework"
+	"volcano.sh/volcano/pkg/scheduler/api"
+	vcache "volcano.sh/volcano/pkg/scheduler/cache"
+	"volcano.sh/volcano/pkg/scheduler/conf"
+	vfwk "volcano.sh/volcano/pkg/scheduler/framework"
+	"volcano.sh/volcano/pkg/scheduler/util"
+)
+
+const (
+	DefaultCandidateNodeCount = 3
+)
+
+type Action struct {
+	fwk *framework.Framework
+	// configured flag for error cache
+	enablePredicateErrorCache bool
+	candidateNodeCount        int
+}
+
+func New() *Action {
+	return &Action{
+		enablePredicateErrorCache: true, // default to enable it
+		candidateNodeCount:        DefaultCandidateNodeCount,
+	}
+}
+
+func (alloc *Action) Name() string {
+	return "allocate"
+}
+
+func (alloc *Action) Initialize() {}
+
+func (alloc *Action) parseArguments(fwk *framework.Framework) {
+	arguments := vfwk.GetArgOfActionFromConf(fwk.Configurations, alloc.Name())
+	arguments.GetBool(&alloc.enablePredicateErrorCache, conf.EnablePredicateErrCacheKey)
+}
+
+func (alloc *Action) Execute(fwk *framework.Framework, schedCtx *agentapi.SchedulingContext) {
+	klog.V(5).Infof("Enter Allocate ...")
+	defer klog.V(5).Infof("Leaving Allocate ...")
+
+	alloc.parseArguments(fwk)
```

Comment:
can you move parseArguments into Initialize, i donot want to see this is run every time